shazbot

killer2k1 opened the pull request.

killer2k1 committed:
antiscum onKilled function fix

Client::onKilled --> AntiScum::onKilled

killer2k1 committed:
client onkilled to antiscum on killed

killer2k1 committed:
createServer function

apparently this needs to be executed before other .cs files

killer2k1 committed:
cancel half countdown if capout occurs

in the event that a capout occurs in the first half, if this happens under 30 seconds when the notifications countdown has started its process, both countdowns will be activated, this update cancels out the notifications countdown so that only the countdown for the second half remains

killer2k1 committed:
suicide time checker update

- made it so suicide time checker does not count observers into the equation.
- changed multiplier value to 0.6 so that time checks every 7 suicides now instead of every 8 suicides.

killer2k1 committed:
observer spawn glitch

- fixed glitch where if you're admin and you join observer while the match has started, it spawns you into thin air
- also fixed bug where if you are on a team while in tourney mode and you select observer it just respawns you
- made it so Automatic team selection option is removed in tourney mode

killer2k1 committed:
Merge branch 'tribesonecommunity:master' into master

killer2k1 committed:
antiscum random messages

added some more fun/playful messages when players get punished by anti-scum

killer2k1 committed:
more precise 30 sec check

at the 40 second mark, schedule another check in 10 seconds, then display the message directly at 30 second mark

killer2k1 committed:
more fixes with 2min warning

in the event a pause occurs within the 3 to 2 minute mark, and the 2 minute warning has been scheduled, this now resets the timer to check again for the 2 minute warning if the paused time exceeds that of the scheduled check time

killer2k1 committed:
Merge branch 'master' into master

killer2k1 committed:
Pause / OOB quality fix update

- made it so when flag carrier exits the mission area, a pause cannot occur, once user re-enters mission area, pause enabled again
- this is to prevent the rare case of pausing the game when someone was OOB with flag (with intention of either throwing it out or bringing it back in)
- if someone pauses the game while someone is OOB with flag, itll be returned by the function of the game.

- fixed small issue in FFA mode where if balance mode was disabled would still recognize the end of the match as the "First half".

killer2k1 committed:
tab to space for code.server.cs

killer2k1 committed:
anti-scum update

- This update prevents flag carriers from self-tossing at the 0-second mark and never taking damage. Grabbing the flag at the 0 second mark will now apply 20% damage upon grab.

killer2k1 committed:
Delete code.antiscum.cs

killer2k1 committed:
anti-scum update

- This solves the issue of self-tossing at the 0-second marker and taking no damage. From now on if the timer reaches 0 seconds, flag carrier that attempts to continue to self-toss will inflict 20% damage upon each grab.

killer2k1 committed:
Various fixes - MA / Dmg Tracking Added

killer2k1 committed:
damage tracking - prevent client side hit sounds

also removed global message when player with flag enters / exits OOB

dwayneyuen committed:
Merge branch 'master' into master

dwayneyuen reviewed (dismissed):
lgtm!

dwayneyuen commented on base/scripts/code.antiscum.cs:
```diff
@@ -395,12 +400,38 @@
+  else { }
+  
+  if (%killedTeam == %killerTeam) {
+      $AntiScum::TeamKilled = true;
```
can you make this a local variable? doesn't seem like it needs to be global

dwayneyuen commented on base/scripts/code.antiscum.cs:
```diff
@@ -395,12 +400,38 @@
+  
+  if (%killedTeam == %killerTeam) {
+      $AntiScum::TeamKilled = true;
+      //we need to set the opposing team value to make the condition valid
+      if (%killedTeam == 1) {
+          %newTeam = 0;
+      }
+      else {
+          %newTeam = 1;
+      }
+  }
```
this can be calculated more concisely as:
```
%newTeam = (%killerTeam + 1) % 2;
```

dwayneyuen commented on base/scripts/code.antiscum.cs:
```diff
@@ -395,12 +400,38 @@
     $AntiScum::lastFlagCarrier[%killerTeam] = "";
   }
+  //death by teammate
+  else if ($AntiScum::TeamKilled && ($AntiScum::lastFlagCarrier[%newTeam] == %playerId)) {
```
good catch here!

dwayneyuen commented on base/scripts/code.antiscum.cs:
```diff
@@ -60,9 +60,9 @@
     Client::sendMessage(%cl, 0, "~wError_Message.wav");
   } else if (%timeLeft <= 0) {
-    if (%timeLeft == 0) {
+    //if (%timeLeft == 0) {
```
is this something people asked for? this new change adds additional text chat spam. the sound still plays below

dwayneyuen commented on base/scripts/code.antiscum.cs:
```diff
@@ -43,13 +43,13 @@
-    Client::sendMessage(%cl, 1, "You have " @ %timeLeft @ " seconds to bring the flag home.~wshell_click.wav");
-    return;
-  }
+  //if (%force) {
```
was this requested by players? i don't know if i agree with this. this informs people when they leave the safe zone and the timer starts. without it, players won't be aware when the timer is ticking.

killer2k1 committed:
antiscum updates

-bring back initial message
-fix double beep sound
-global teamkilled variables set to local
